src/map.js

```diff
@@ -0,0 +1,233 @@
+  quitGame() {
+    this.setState({ gameStarted: this });
+    if (this.state.gameStarted) {
+      this.map.fitWorld([44.0286, -72.7317], 10);
+    }
+  }
+
+  // checkVermontLocation() {
+  //    { randomLat, randomLong } = this.state;
+  // }
+
+  randomLat() {
+    let lat =
+      Math.random() * (45.00541896831666 - 42.730315121762715) +
+      42.730315121762715;
+    return lat;
+  }
+  randomLong() {
+    let long =
+      Math.random() * (71.51022535353107 - 73.35218221090553) +
+      73.35218221090553 * -1;
+    return long;
+  }
+
+  moveMapNorth() {
+    const { lat, lng } = this.state.currentPoint;
+    this.setState({
+      currentPoint: {
+        lat: lat + 0.002,
+        lng: lng + 0.0
+      }
+    });
+  }
+
+  moveMapSouth() {
+    const { lat, lng } = this.state.currentPoint;
+    this.setState({
+      currentPoint: {
+        lat: lat - 0.002,
+        lng: lng + 0.0
+      }
+    });
+  }
+  moveMapEast() {
+    const { lat, lng } = this.state.currentPoint;
+    this.setState({
+      currentPoint: {
+        lat: lat + 0.0,
+        lng: lng + 0.002
+      }
+    });
+  }
+  moveMapWest() {
+    const { lat, lng } = this.state.currentPoint;
+    this.setState({
+      currentPoint: {
+        lat: lat + 0.0,
+        lng: lng - 0.002
+      }
+    });
+  }
+
+  componentDidMount() {
+    document.querySelector("#score").textContent = `Your Score: 0`;
+    this.map = L.map("map", {
+      center: [this.state.currentPoint.lat, this.state.currentPoint.lng],
+      // maxBounds: [44.0886, -72.7317],
+      zoom: this.state.defaultZoom,
+      layers: [
+        L.tileLayer(
+          "https://server.arcgisonline.com/ArcGIS/rest/services/World_Imagery/MapServer/tile/{z}/{y}/{x}",
+          {
+            attribution:
+              "Tiles &copy; Esri &mdash; Source: Esri, i-cubed, USDA, USGS, AEX, GeoEye, Getmapping, Aerogrid, IGN, IGP, UPR-EGP, and the GIS User Community"
+          }
+        )
+      ]
+    });
+
+    this.marker = L.marker(this.state.markerPosition).addTo(this.map);
+    this.borderData = L.geoJSON(borderData);
+    this.borderData.addTo(this.map);
+    const bounds = L.bounds(44.0886, -72.7317);
+    this.map.setMinZoom(this.state.minimumZoom);
+    this.map.setMaxZoom(this.state.maximumZoom);
+    let layerLength = leafletPip.pointInLayer(
+      this.state.currentPoint,
+      this.borderData
+    ).length;
+    while (!layerLength) {
+      let randomLat = this.randomLat();
+      let randomLong = this.randomLong();
+      this.setState({
+        randomLat: randomLat,
+        randomLong: randomLong
+      });
+      layerLength = leafletPip.pointInLayer(
+        [this.state.randomLong, this.state.randomLat],
+        this.borderData
+      ).length;
+    }
+  }
+
+  componentDidUpdate() {
+    let { markerPosition, oldMarkerPosition, currentPoint } = this.state;
+    if (markerPosition.lat !== markerPosition.oldLat) {
```

Comment:
Just check against `markerPostion` rather than `markerPosition.lat` to allow E-W movement as well as N-S.